xymon-snmpcollect build fails when Net-SNMP MD5 auth symbol is unavailable

`xymon-snmpcollect` fails to compile on openSUSE Tumbleweed because `usmHMACMD5AuthProtocol` is missing.

Error:
`xymonnet/xymon-snmpcollect.c:429:55: error: 'usmHMACMD5AuthProtocol' undeclared`

Expected: if MD5 is not available in Net-SNMP, build should still pass (fallback to SHA1).

--- a/xymonnet/xymon-snmpcollect.c
+++ b/xymonnet/xymon-snmpcollect.c
@@ -248,6 +248,7 @@ int print_result (int status, req_t *req, struct snmp_pdu *pdu)
 }
 
 
+
 /*
  * response handler
  */
@@ -426,12 +427,19 @@ void startonehost(struct req_t *req, int ipchange)
 			/* set the authentication method */
 			switch (req->authmethod) {
 			  case SNMP_V3AUTH_MD5:
+#ifndef NETSNMP_DISABLE_MD5
 				s.securityAuthProto = usmHMACMD5AuthProtocol;
 				s.securityAuthProtoLen = sizeof(usmHMACMD5AuthProtocol)/sizeof(oid);
 				s.securityAuthKeyLen = USM_AUTH_KU_LEN;
 				break;
+#else
+				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support. Falling back to SHA1.\n",
+					  req->hostname);
+				/* FALLTHROUGH */
+#endif
 
 			  case SNMP_V3AUTH_SHA1:
+			  default:
 				s.securityAuthProto = usmHMACSHA1AuthProtocol;
 				s.securityAuthProtoLen = sizeof(usmHMACSHA1AuthProtocol)/sizeof(oid);
 				s.securityAuthKeyLen = USM_AUTH_KU_LEN;
@@ -686,7 +694,11 @@ void readconfig(char *cfgfn, int verbose)
 
 			reqitem->hostip[0] = reqitem->hostname;
 			reqitem->version = SNMP_VERSION_1;
+#ifdef NETSNMP_DISABLE_MD5
+			reqitem->authmethod = SNMP_V3AUTH_SHA1;
+#else
 			reqitem->authmethod = SNMP_V3AUTH_MD5;
+#endif
 			reqitem->next = reqhead;
 			reqhead = reqitem;
 
@@ -732,8 +744,15 @@ void readconfig(char *cfgfn, int verbose)
 		}
 
 		if (strncmp(bot, "authmethod=", 11) == 0) {
-			if (strcasecmp(bot+11, "md5") == 0)
+			if (strcasecmp(bot+11, "md5") == 0) {
+#ifdef NETSNMP_DISABLE_MD5
+				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support. Using SHA1.\n",
+					  reqitem->hostname);
+				reqitem->authmethod = SNMP_V3AUTH_SHA1;
+#else
 				reqitem->authmethod = SNMP_V3AUTH_MD5;
+#endif
+			}
 			else if (strcasecmp(bot+11, "sha1") == 0)
 				reqitem->authmethod = SNMP_V3AUTH_SHA1;
 			else

--- a/xymonnet/xymon-snmpcollect.c
+++ b/xymonnet/xymon-snmpcollect.c
@@ -433,9 +433,10 @@ void startonehost(struct req_t *req, int ipchange)
 				s.securityAuthKeyLen = USM_AUTH_KU_LEN;
 				break;
 #else
-				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support. Falling back to SHA1.\n",
+				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support.\n",
 					  req->hostname);
-				/* FALLTHROUGH */
+				xfree(s.securityName);
+				return;
 #endif
 
 			  case SNMP_V3AUTH_SHA1:
@@ -746,12 +747,10 @@ void readconfig(char *cfgfn, int verbose)
 		if (strncmp(bot, "authmethod=", 11) == 0) {
 			if (strcasecmp(bot+11, "md5") == 0) {
 #ifdef NETSNMP_DISABLE_MD5
-				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support. Using SHA1.\n",
+				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support.\n",
 					  reqitem->hostname);
-				reqitem->authmethod = SNMP_V3AUTH_SHA1;
-#else
-				reqitem->authmethod = SNMP_V3AUTH_MD5;
 #endif
+				reqitem->authmethod = SNMP_V3AUTH_MD5;
 			}
 			else if (strcasecmp(bot+11, "sha1") == 0)
 				reqitem->authmethod = SNMP_V3AUTH_SHA1;
@@ -1132,4 +1131,3 @@ int main (int argc, char **argv)
 
 	return 0;
 }
-

--- a/xymonnet/xymon-snmpcollect.c
+++ b/xymonnet/xymon-snmpcollect.c
@@ -426,12 +426,20 @@ void startonehost(struct req_t *req, int ipchange)
 			/* set the authentication method */
 			switch (req->authmethod) {
 			  case SNMP_V3AUTH_MD5:
+#ifndef NETSNMP_DISABLE_MD5
 				s.securityAuthProto = usmHMACMD5AuthProtocol;
 				s.securityAuthProtoLen = sizeof(usmHMACMD5AuthProtocol)/sizeof(oid);
 				s.securityAuthKeyLen = USM_AUTH_KU_LEN;
 				break;
+#else
+				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support.\n",
+					  req->hostname);
+				xfree(s.securityName);
+				return;
+#endif
 
 			  case SNMP_V3AUTH_SHA1:
+			  default:
 				s.securityAuthProto = usmHMACSHA1AuthProtocol;
 				s.securityAuthProtoLen = sizeof(usmHMACSHA1AuthProtocol)/sizeof(oid);
 				s.securityAuthKeyLen = USM_AUTH_KU_LEN;
@@ -686,7 +694,11 @@ void readconfig(char *cfgfn, int verbose)
 
 			reqitem->hostip[0] = reqitem->hostname;
 			reqitem->version = SNMP_VERSION_1;
+#ifdef NETSNMP_DISABLE_MD5
+			reqitem->authmethod = SNMP_V3AUTH_SHA1;
+#else
 			reqitem->authmethod = SNMP_V3AUTH_MD5;
+#endif
 			reqitem->next = reqhead;
 			reqhead = reqitem;
 
@@ -732,8 +744,13 @@ void readconfig(char *cfgfn, int verbose)
 		}
 
 		if (strncmp(bot, "authmethod=", 11) == 0) {
-			if (strcasecmp(bot+11, "md5") == 0)
+			if (strcasecmp(bot+11, "md5") == 0) {
+#ifdef NETSNMP_DISABLE_MD5
+				errprintf("SNMPv3 authmethod MD5 requested for host %s, but Net-SNMP was built without MD5 support.\n",
+					  reqitem->hostname);
+#endif
 				reqitem->authmethod = SNMP_V3AUTH_MD5;
+			}
 			else if (strcasecmp(bot+11, "sha1") == 0)
 				reqitem->authmethod = SNMP_V3AUTH_SHA1;
 			else
@@ -1113,4 +1130,3 @@ int main (int argc, char **argv)
 
 	return 0;
 }
-

